patches/src/main/kotlin/app/revanced/patches/spotify/misc/UnlockPremiumPatch.kt

```diff
@@ -62,47 +66,82 @@ val unlockPremiumPatch = bytecodePatch(
+            )
+
+        }
+
         // Make featureTypeCase_ accessible so we can check the home section type in the extension.
         homeSectionFingerprint.classDef.fields.first { it.name == "featureTypeCase_" }.apply {
             accessFlags = accessFlags.or(AccessFlags.PUBLIC.value).and(AccessFlags.PRIVATE.value.inv())
-        }
 
-        val protobufListClassName = with(protobufListsFingerprint.originalMethod) {
-            val emptyProtobufListGetIndex = indexOfFirstInstructionOrThrow(Opcode.SGET_OBJECT)
-            getInstruction(emptyProtobufListGetIndex).getReference<FieldReference>()!!.definingClass
-        }
 
-        val protobufListRemoveFingerprint = fingerprint {
-            custom { method, classDef ->
-                method.name == "remove" && classDef.type == protobufListClassName
+            val protobufListClassName = with(protobufListsFingerprint.originalMethod) {
+                val emptyProtobufListGetIndex = indexOfFirstInstructionOrThrow(Opcode.SGET_OBJECT)
+                getInstruction(emptyProtobufListGetIndex).getReference<FieldReference>()!!.definingClass
             }
-        }
 
-        // Need to allow mutation of the list so the home ads sections can be removed.
-        // Protobuffer list has an 'isMutable' boolean parameter that sets the mutability.
-        // Forcing that always on breaks unrelated code in strange ways.
-        // Instead, remove the method call that checks if the list is unmodifiable.
-        with(protobufListRemoveFingerprint.method) {
-            val invokeThrowUnmodifiableIndex = indexOfFirstInstructionOrThrow {
-                val reference = getReference<MethodReference>()
-                opcode == Opcode.INVOKE_VIRTUAL &&
-                        reference?.returnType == "V" && reference.parameterTypes.isEmpty()
+            val protobufListRemoveFingerprint = fingerprint {
+                custom { method, classDef ->
+                    method.name == "remove" && classDef.type == protobufListClassName
+                }
             }
 
-            // Remove the method call that throws an exception if the list is not mutable.
-            removeInstruction(invokeThrowUnmodifiableIndex)
-        }
+            // Need to allow mutation of the list so the home ads sections can be removed.
+            // Protobuffer list has an 'isMutable' boolean parameter that sets the mutability.
+            // Forcing that always on breaks unrelated code in strange ways.
+            // Instead, remove the method call that checks if the list is unmodifiable.
+            with(protobufListRemoveFingerprint.method) {
+                val invokeThrowUnmodifiableIndex = indexOfFirstInstructionOrThrow {
+                    val reference = getReference<MethodReference>()
+                    opcode == Opcode.INVOKE_VIRTUAL &&
+                            reference?.returnType == "V" && reference.parameterTypes.isEmpty()
```

Comment:
The opcode check may not be necessary:

```suggestion
                    val reference = getReference<MethodReference>()
                    	?: return@indexOfFirstInstructionOrThrow null
                    	
                    reference.returnType == "V" && reference.parameterTypes.isEmpty()
```

if the opcode is required then keep it and use the null check operator